capstone/src/main/java/com/google/step/servlets/LoginServlet.java

```diff
@@ -0,0 +1,103 @@
+import com.google.appengine.api.datastoreDatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserserviceFactory();
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.Map;
+import java.util.HashMap;
+import com.google.gson.Gson;
+
+/**
+ * Servlet that will handle login information.
+ */
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+
+        UserService userService = UserServiceFactory.getUserService();
+        Map<String, Object> login = new HashMap<>();
+
+        if (userService.isLoggedIn()) {
+            String useremail = userService.getCurrentUser().getEmail();
+            String type = checkIfRestaurantOrUser(useremail);
+
+            //If user is loggedin but not in our system.
+            if (type == null) {
+                response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+                return;
+            }
+
+            String urltoRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urltoRedirectToAfterUserLogsOut);
+
+            login.put("Loggedin", true);
+            login.put("Type", type);
+            login.put("Email", useremail);
+            login.put("logOutUrl", logoutUrl);
+        } else {
+
+            String urltoRedirectToAfterUserLogIn = "/";
+            Stromg urltoRedirectAfterUserSignUp = "/user-signup.html";
```

Comment:
Nit: make this "String" instead of "Stromg"

Reply:
Done... whoops